modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx

```diff
@@ -9,34 +9,155 @@ import {ClayInput} from '@clayui/form';
 import ClayLayout from '@clayui/layout';
 import ClayTable from '@clayui/table';
 import classNames from 'classnames';
-import {openModal} from 'frontend-js-web';
-import React from 'react';
+import {fetch, openModal} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
 
 import '../../../../css/ListVisualizationMode.scss';
 import {IFDSViewSectionProps} from '../../../FDSView';
+import FieldSelectModalContent from '../../../components/FieldSelectModalContent';
+import {API_URL, OBJECT_RELATIONSHIP} from '../../../utils/constants';
+import openDefaultFailureToast from '../../../utils/openDefaultFailureToast';
+import openDefaultSuccessToast from '../../../utils/openDefaultSuccessToast';
+import {IField} from '../../../utils/types';
 import {IBaseVisualizationMode} from '../VisualizationModes';
-import {IFDSField} from '../table/Table';
-import AddFieldsModalContent from '../table/modal_content/AddFieldsModalContent';
 
 export interface IList extends IBaseVisualizationMode<'list'> {}
-
-interface IListField {
-	fieldId: string;
+interface IFDSListSection {
+	externalReferenceCode: string;
+	fieldName: string;
+	name: string;
+	rendererName?: string;
+}
+interface IListSection {
+	externalReferenceCode?: IFDSListSection['externalReferenceCode'];
+	field?: IField;
 	label: string;
+	name: IFDSListSection['name'];
```

Comment:
- `IFDSListSection` represents the stored entity.
- `IListSection` represents the visual representation of the stored entity.